app/views/admin/roles/_form.html.erb

```diff
@@ -109,7 +109,7 @@
       rows: 20
     } %>
 
-    <div class="govuk-button-group">
+    <div class="govuk-button-group" data-module="track-button-click" data-track-category="form-button" data-track-action=<%= "#{form.object.class.name.demodulize.underscore.dasherize}-button" %>>
       <%= render "govuk_publishing_components/components/button", {
```

Comment:
so here both save buttons trigger slightly different events 

<img width="439" alt="image" src="https://user-images.githubusercontent.com/42515961/228891233-c8286375-b0a2-40a0-a4c4-169a5a67e496.png">

<img width="448" alt="image" src="https://user-images.githubusercontent.com/42515961/228891477-92b31e30-3462-4011-bf85-92d8de5a2051.png">

so we can do 

```
  <div class="govuk-button-group">
      <%= render "govuk_publishing_components/components/button", {
        text: "Save",
        name: "save",
        value: "Save",
        type: "submit",
        data_attributes: {
          module: "gem-track-click",
          "track-category": "form-button",
          "track-action": "#{role.class.name.demodulize.underscore.dasherize}-button",
          "track-label": "Save"
        }
      } %>

      <%= render "govuk_publishing_components/components/button", {
        text: "Save and continue",
        name: "save_and_continue",
        value: "Save and continue",
        type: "submit",
        secondary_solid: true,
        data_attributes: {
          module: "gem-track-click",
          "track-category": "form-button",
          "track-action": "#{role.class.name.demodulize.underscore.dasherize}-button",
          "track-label": "Save and continue"
        }
      } %>
      <%= link_to("Cancel", admin_roles_path, class: "govuk-link") %>
    </div>
```

Reply:
answered here https://gds.slack.com/archives/C03R56148PR/p1680193176378349?thread_ts=1680188824.037199&cid=C03R56148PR but the tldr is

`gem-track-click` for design system pages and `track-button-click` for pages built in Bootstrap